### DISCUSSION TITLE
Hubs-Cloud: Question about the configuration for an event.

### DISCUSSION THREAD

**Discussion by elmau** [mozilla/hubs#2942] [Hubs Cloud Questions] [answered]
Hi, recently we were asked to set up an event using hubs-cloud. It's a concert, and we would like to know what's the best configuration for an event. The expected ccu is around 5000 simultaneous. We don't know if they'll be in different rooms or in the lobby.

**We would like to know a couple of things:** 

- If is there a way to separate the access for the people that are going to attend to go in-room, like a VIP pass and the people who will only see from the lobby? We're thinking in three types of configurations: 1) a few rooms with VIP passes and the rest watch from the lobby. 2) vip passes for the webvr experience, and the rest watch from another website with only video streaming and no hubs. 3) Or a lot of rooms to allow the full required capacity of the event. We're still evaluating and we'll choose according to budget and recommendations.

- Also in the case of having a lot of rooms with 25-30 people in them. The event has a common stream for every room. Is it better to stream it from an external source, or having the video pre-loaded and running in the scene? How this affect server costs?

Thank you in advance.

**Comment by emclaren** [comment, marked as answer]
Hi Elmau - thanks for reaching out. This sounds like an exciting event!

I will try my best to answer your questions. 

* First off - to my knowledge there has never been a single room with thousands of people watching from the lobby. We've tested up to several hundred bots watching from the lobby successfully. This is not to say that it is not possible, however, it's not a common use case and its not something we've tested. 

* You can separate out access to different places in the room using special flag on the URL (for more information see: https://github.com/mozilla/hubs/pull/2918); however, there is no way to send some people into the room and other people to the lobby. 

Entrance to the room is based on a first come - first served basis. That being said room moderators can always enter a room even if it is full - so you could always set the room capacity in the room settings to be 0, and make every VIP person a moderator in that particular room. This would be a bit of a challenge to set up as you would need to actually enter the room with every VIP attendee and promote them manually (there is no automated way to do this in the back end). 

Streaming to a different platform might be a better alternative for the folks you want to be able to see the experience without entering into the room as you wouldn't need to worry about promoting VIPs, and you wouldn't have the an untested number of folks watching from the lobby. We have shared some tips on recording in Hubs https://blog.mozvr.com/recording-inside-of-hubs/ If you have questions on streaming from Hubs feel free to open a new discussion topic. 

* I'm not totally sure what you mean in your second question. Would you be able to give me a bit more detail?  

You can set up a common stream from a platform like twitch to your different Hubs rooms. You can possibly even set them up in a scene in Spoke before the stream starts (this way you wouldnt have to manually put the stream into every room), however, I would recommend testing this in advance as it's not frequently done (PS if you do any testing on this, please let us know if you experience any bugs). I don't believe pre-loading a video hosted on another server will have any impact on server costs, so long as no one is in a room with the video it won't be playing & using bandwidth in advance of the event (Note that if you have a pre-recorded video that you host on your Hubs Cloud server you will need to pay for the storage of the video).

**Reply by elmau** [reply]
Hi Elgin,

thank you for the answers!. Regarding my second question. We're trying to figure out the best way to stream an event: it's a dj set, so streaming is tricky in most platforms because of copyright. 

There's one thing we are still testing. And we failed with the tests: that going inside a room with a preloaded video, the video doesn't sync between users. (or at least the lag is very evident). And in another test, one user entered and saw the video, and a second user entered later and saw the video from the beginning. So we need that the video is at least running with one user inside and when the others enter the room, see that the video is already running.

We think that streaming live from an external source solves this, however we're not sure if it's going to work properly.
